apps/nextjs/src/app/api/chat/user.ts

```diff
@@ -2,108 +2,162 @@ import { auth, clerkClient } from "@clerk/nextjs/server";
 import { ErrorDocument } from "@oakai/aila/src/protocol/jsonPatchProtocol";
 import { demoUsers, inngest } from "@oakai/core";
 import { posthogServerClient } from "@oakai/core/src/analytics/posthogServerClient";
+import { withTelemetry } from "@oakai/core/src/tracing/serverTracing";
 import { rateLimits } from "@oakai/core/src/utils/rateLimiting/rateLimit";
 import { RateLimitExceededError } from "@oakai/core/src/utils/rateLimiting/userBasedRateLimiter";
+import invariant from "tiny-invariant";
 
 import { streamingJSON } from "./protocol";
 
+export async function handleUserLookup(chatId: string) {
+  return await withTelemetry(
+    "chat-user-lookup",
+    { chat_id: chatId },
+    async (userLookupSpan) => {
+      invariant(chatId, "Chat ID is required");
+
+      const result = await fetchAndCheckUser(chatId);
+      if (!result) {
+        userLookupSpan.setTag("error", true);
+        userLookupSpan.setTag("error.message", "user lookup failed");
+        return {
+          failureResponse: new Response("User lookup failed", {
+            status: 400,
+          }),
+        };
+      }
+      if ("failureResponse" in result) {
+        userLookupSpan.setTag("error", true);
+        userLookupSpan.setTag("error.message", "user lookup failed");
```

Comment:
In a future PR I wonder if withTelemetry could provide a function to record an error to avoid some of this noise